src/Tools/ExternalAccess/Extensions/External/ExtensionMessageContext.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Extensions;
+
+/// <summary>
+/// Represents the context of an extension message handler.
+/// </summary>
+public sealed class ExtensionMessageContext
```

Comment:
why do we need this?  why not just pass Solution?

Reply:
We should keep a message context.  We may need to pass other information in the future (e.g. getting services for example).